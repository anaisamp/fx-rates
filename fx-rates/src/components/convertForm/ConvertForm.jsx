import React from 'react';
import currencies from '../../constants';

const ConvertForm = ({ selectName, selectedCurrency, inputValue, handleCurrencyChange, handleExchangeRate }) => (
  <div>
    <select name={selectName} value={selectedCurrency} onChange={handleCurrencyChange}>
      { currencies.map(currency =>
        <option value={currency.symbol} key={currency.symbol}>
          {currency.symbol}
        </option>)
      }
    </select>
    <input
      type="number"
      step="0.01"
      name={selectedCurrency}
      value={inputValue}
      onChange={handleExchangeRate}
    />
  </div>
);

export default ConvertForm;