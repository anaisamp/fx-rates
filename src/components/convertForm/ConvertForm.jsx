import React, { Component } from 'react';
import PropTypes from 'prop-types';
import currencies from '../../constants/currencies';
import './convertForm.css';

class ConvertForm extends Component {
  componentDidMount() {
    this.focusFromInput();
  }

  focusFromInput = () => {
    if (this.props.selectName === 'from' && this.fromInput) {
      this.fromInput.focus();
    }
  }

  render() {
    const {
      selectName, selectedCurrency, inputValue, handleCurrencyChange, handleExchangeRate, balance,
    } = this.props;
    return (
      <div className="convert-form">
        <div className="convert-wrapper">
          <span className="select-wrapper">
            <select name={selectName} value={selectedCurrency} onChange={handleCurrencyChange}>
              { currencies.map(currency => (
                <option value={currency.symbol} key={currency.symbol}>
                  {currency.symbol}
                </option>
              ))
          }
            </select>
          </span>
          <input
            type="number"
            step="0.01"
            name={selectedCurrency}
            value={inputValue}
            onChange={handleExchangeRate}
            placeholder="0"
            min="0"
            ref={(input) => { this.fromInput = input; }}
          />
        </div>
        <p className="balance">
          Balance:
          {' '}
          {currencies.filter(c => c.symbol === selectedCurrency)[0].currency}
          {parseFloat(balance).toFixed(2)}
        </p>
      </div>
    );
  }
}
ConvertForm.propTypes = {
  selectName: PropTypes.string.isRequired,
  selectedCurrency: PropTypes.string.isRequired,
  inputValue: PropTypes.string.isRequired,
  handleCurrencyChange: PropTypes.func.isRequired,
  handleExchangeRate: PropTypes.func.isRequired,
  balance: PropTypes.string.isRequired,
};

export default ConvertForm;
