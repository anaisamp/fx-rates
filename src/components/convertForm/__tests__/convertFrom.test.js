import React from 'react';
import { shallow } from 'enzyme';
import '../../../config/enzyme-setup';
import ConvertForm from '../ConvertForm';

it('renders ConvertForm', () => {
  const props = {
    selectName: 'from',
    selectedCurrency: 'USD',
    inputValue: '11',
    handleCurrencyChange: jest.fn(),
    handleExchangeRate: jest.fn(),
    balance: '30.00',
  };
  const wrapper = shallow(<ConvertForm {...props} />);
  expect(wrapper).toMatchSnapshot();
});
