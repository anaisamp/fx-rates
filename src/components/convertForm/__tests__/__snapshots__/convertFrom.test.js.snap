// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ConvertForm 1`] = `
<div
  className="convert-form"
>
  <div
    className="convert-wrapper"
  >
    <span
      className="select-wrapper"
    >
      <select
        name="from"
        onChange={[Function]}
        value="USD"
      >
        <option
          key="GBP"
          value="GBP"
        >
          GBP
        </option>
        <option
          key="EUR"
          value="EUR"
        >
          EUR
        </option>
        <option
          key="USD"
          value="USD"
        >
          USD
        </option>
      </select>
    </span>
    <input
      min="0"
      name="USD"
      onChange={[Function]}
      placeholder="0"
      step="0.01"
      type="number"
      value="11"
    />
  </div>
  <p
    className="balance"
  >
    Balance:
     
    $
    30.00
  </p>
</div>
`;
